### DISCUSSION TITLE
Is it possible to output the json payload to console?

### DISCUSSION THREAD

**Discussion by SimonBalling** [serilog-contrib/serilog-sinks-grafana-loki#239] [Q&A]
Hi,

I want to output the payload that is sent to GrafanaLoki also to the console. Is this possible?

If I could intercept the message and output it myself

**Comment by mishamyte** [comment]
Hi @SimonBalling!

Usually you just configure multiple sinks for Serilog (for example `Console` + `Grafana.Loki`)

**Comment by SimonBalling** [comment]
I know. But I want to get the exact same output in both

**Reply by mishamyte** [reply]
Hmmm, can't tell you the way and have some doubts it's possible.

Previously, some people just bypassed the `LokiJsonTextFormatter ` into the `Console` sink, but it won't be exactly the same (it will be just the payload formatting).

**Comment by MitoG** [comment]
In case there is still a need I did it in a rather dirty way by making a custom LokiHttpClient

```
public class CustomLokiHttpClient : LokiHttpClient
{
    public override Task<HttpResponseMessage> PostAsync(string requestUri, Stream contentStream)
    {
        using (var s = new MemoryStream())
        {
            contentStream.CopyTo(s);
            var bytes = s.ToArray();
            var output = Encoding.ASCII.GetString(bytes);            
        }
        
        return base.PostAsync(requestUri, contentStream);
    }
}
```
